test/extensions/filters/udp/dns_filter/dns_filter_integration_test.cc

```diff
@@ -236,7 +235,7 @@ TEST_P(DnsFilterIntegrationTest, ExternalLookupTest) {
       Utils::buildQueryForDomain("www.google.com", DNS_RECORD_TYPE_A, DNS_RECORD_CLASS_IN);
   requestResponseWithListenerAddress(*listener_address, query, response);
 
-  query_ctx_ = response_parser_->createQueryContext(response, counters_);
+  query_ctx_ = response_parser_->createResponseContext(response, counters_);
```

Comment:
I think the test code should test the production code, which is createQueryContext(). The test code should not test the testing specific code logic.  Same as all below.

Reply:
Right, the response is already got from the DNS filter, which is from `requestResponseWithListenerAddress(*listener_address, query, response)`. This line is used to verify the response generated by the DNS filter.